lib/FIR/statements.h

```diff
@@ -378,25 +401,26 @@ class LoadInsn : public MemoryStmt_impl {
 // Store value(s) from an applied expression to a location
 class StoreInsn : public MemoryStmt_impl {
 public:
-  using ValueType =
-      std::variant<Value, ApplyExprStmt *, Addressable_impl *, BasicBlock *>;
-  template<typename T> static StoreInsn Create(T *addr, T *value) {
-    return StoreInsn{addr, value};
-  }
-  template<typename T> static StoreInsn Create(T *addr, BasicBlock *value) {
+  using ValueType = std::variant<Value, QualifiedStmt<ApplyExprStmt>,
+      QualifiedStmt<Addressable_impl>, BasicBlock *>;
+
+  template<typename A>
+  static StoreInsn Create(QualifiedStmt<Addressable_impl> addr, A value) {
     return StoreInsn{addr, value};
   }
 
   Addressable_impl *address() const { return address_; }
   ValueType value() const { return value_; }
 
 private:
-  explicit StoreInsn(Value addr, Value val);
-  explicit StoreInsn(Value addr, BasicBlock *val);
-  explicit StoreInsn(Statement *addr, Statement *val);
-  explicit StoreInsn(Statement *addr, BasicBlock *val);
-
-  Addressable_impl *address_;
+  explicit StoreInsn(QualifiedStmt<Addressable_impl> addr, Value val);
+  explicit StoreInsn(
+      QualifiedStmt<Addressable_impl> addr, QualifiedStmt<ApplyExprStmt> val);
```

Comment:
Formatting for this line and the next one seems different.

Reply:
I'll re-run clang-format on this.